terratest/utils/extensions.go

```diff
@@ -449,7 +449,7 @@ func (i *Instance) waitForApp(predicate func(instances int) bool, stop bool) (er
 	i.w.t.Logf("Wait for coreDNS to be filled by local targets %s", i.w.state.gslb.host)
 	for n := 0; n < maxRetries/2; n++ {
 		localTargets := i.GetLocalTargets()
-		if len(localTargets) == 0 {
+		if (!stop && len(localTargets) == 0) || (stop && len(localTargets) != 0) {
```

Comment:
This is a follow up of https://github.com/k8gb-io/k8gb/pull/1682 since the number of `localTargets` should be 0 if the action is stop. It helps keeping the code more understandable but it didn't cause any issues since in that case the code would return already on line 431.